app/controllers/application_controller.rb

```diff
@@ -1,2 +1,19 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    added_attrs = [:username, :email, :password, :password_confirmation, :remember_me]
+    devise_parameter_sanitizer.permit :sign_up, keys: added_attrs
+    devise_parameter_sanitizer.permit :sign_in, keys: [:login, :password]
+    devise_parameter_sanitizer.permit :account_update, keys: [:content, added_attrs]
```

Comment:
If I remember correctly, we put only additional fields in `devise_parameter_sanitizer.permit`, and devise handles `:email, :password, :password_confirmation, :remember_me` by default, it's only the username that we're adding. So, we need additional filed of `:username` during signup and `:username` and `:content` (btw waht is content?) during `account_update`

Reply:
I add username to added_attrs = [:username, :email, :password, :password_confirmation, :remember_me] so thats how i add it to signup. And I made user to be able to sign in with :email or :username
And content is  an action text - rich_text_area, with that I add the opportunity to the user to upload a picture when he edits his profile

Reply:
I mean that devise already knows that you need email and password to sign up, we don't need to add them. See here https://www.rubydoc.info/github/plataformatec/devise/Devise/ParameterSanitizer

Reply:
```
class ApplicationController < ActionController::Base
  before_action :authenticate_user!
  before_action :configure_permitted_parameters, if: :devise_controller?
  

  protected

  def configure_permitted_parameters    
    devise_parameter_sanitizer.permit :sign_up, keys: [:username, :email]    
    devise_parameter_sanitizer.permit :account_update, keys: [:content, :username, :email]
  end
```

Now I did that using the default devise parameters. Allowing to sign up with username or email(I add email in sign up because there was some problem when initialize new sign up, wasn't working). And in account update there will be possibility to change username, email and add a picture